src/main/haskell/kore/src/Kore/OnePath/Verification.hs

```diff
@@ -235,5 +237,93 @@ verifyClaim
         StrategyPattern.Stuck p : _ -> throwE p
         StrategyPattern.Bottom : _ -> error "Unexpected bottom pattern."
   where
-    liftIO' :: IO () -> ExceptT (CommonExpandedPattern level) Simplifier ()
-    liftIO' = liftIO
+    transitionRule'
+        :: Prim (CommonExpandedPattern level) (RewriteRule level Variable)
+        -> (StrategyPattern (CommonExpandedPattern level), StepProof level Variable)
+        -> Simplifier [(StrategyPattern (CommonExpandedPattern level), StepProof level Variable)]
+    transitionRule' =
+        OnePath.transitionRule
+            metadataTools
+            substitutionSimplifier
+            simplifier
+            axiomIdToSimplifier
+
+-- | TODO: Docs.
+type Configuration level = StrategyPattern (CommonExpandedPattern level)
+
+
+-- | Attempts to perform a single proof step, starting at the configuration
+-- in the execution graph designated by the provided node. Re-constructs the
+-- execution graph by inserting this step.
+verifyClaimStep
+    :: forall level
+    .  MetaOrObject level
+    => MetadataTools level StepperAttributes
+    -> StepPatternSimplifier level
+    -> PredicateSubstitutionSimplifier level
+    -> BuiltinAndAxiomSimplifierMap level
+    -> Claim level
+    -- ^ claim that is being proven
+    -> [Claim level]
+    -- ^ list of claims in the spec module
+    -> [Axiom level]
+    -- ^ list of axioms in the main module
+    -> ExecutionGraph (Configuration level)
+    -- ^ current execution graph
+    -> Graph.Node
+    -- ^ selected node in the graph
+    -> Simplifier (ExecutionGraph (Configuration level))
+verifyClaimStep
+    tools
+    simplifier
+    predicateSimplifier
+    axiomIdToSimplifier
+    target
+    claims
+    axioms
+    eg@ExecutionGraph { root }
+    node
+  = executionHistoryStep
+        transitionRule'
+        strategy'
+        eg
+        node
+
+  where
+    transitionRule'
+        :: Prim (CommonExpandedPattern level) (RewriteRule level Variable)
+        -> StrategyPattern (CommonExpandedPattern level)
+        -> Simplifier [StrategyPattern (CommonExpandedPattern level)]
+    transitionRule' =
+        \prim ->
```

Comment:
It feels to me that this was written to be short, not readable. Please consider whether adding function arguments for the function and writing do-style code would make it more readable.